Feature/205 add pagination

KC900201 opened the pull request:
# Include paginatio logic to home page

Improvements completed:
- Add in pagination logic with options of 6, 12, 24, 48 cards to display (`6` is default)
- Include previous and next button to allow user navigating through results
- Include test scripts for pagination logic

## Screenshot:
### Before
<img width="1906" height="986" alt="image" src="https://github.com/user-attachments/assets/09977026-8e3d-4b89-a290-19b6f2e6e0b1" />

### After
<img width="1917" height="967" alt="image" src="https://github.com/user-attachments/assets/2cb8d1de-b1b7-4eb1-8ba3-b7896be9247a" />

KC900201 commented on src/scripts/paginate.ts:
```diff
@@ -0,0 +1,140 @@
+    })
+  } else {
+    new PaginationManager()
+  }
```
Removed line 129 - 130 to avoid duplication

Buzzpy commented:
Hi @KC900201
Thanks a bunch for the contribution! I'll review it as soon as possible and get back to you. Have a great day ahead!

KC900201 commented:
Hi @Buzzpy, happy new year. May I know how is the PR review progress going?

Buzzpy commented:
Great work @KC900201! And extremely sorry for the late reply. 
I took a look at the PR and have some feedback:
- We can increase the items per page to be 10, 20, 30, and 40— makes it simple and generalized.
- Can we slightly decrease this padding? I understand it might have been more relevant to the pagination bar, but it seems like a lot of dead space.

<img width="990" height="490" alt="image" src="https://github.com/user-attachments/assets/9671221a-f925-4738-8f2d-1b0714d6930b" />
- On the mobile, the bar looks like below (flexbox? I'm not sure!):  
<img width="400" height="691" alt="image" src="https://github.com/user-attachments/assets/fd333f37-aa5e-4775-beef-f373c45460cb" />

We can perhaps just center it— easier, but it may look bulky given the other tabs are already centered, or just put it all together on the right as below:  
<img width="350" height="501" alt="image" src="https://github.com/user-attachments/assets/1ba83743-64c8-4817-b2a4-d8506d0ac73b" />

Let me know your thoughts as well! And I apologize if I'm making some absurd statements, you already know my not-so-amazing capabilities in web design by now >.<

And if you can not find time for these fixes— which is very understandable given the timelines— let me know. I might be able to make some changes as well.

KC900201 commented:
@Buzzpy, 

Thanks for the feedback. I've updated the code according to your feedback. Kindly review.

Updated the pagination option value
<img width="257" height="233" alt="image" src="https://github.com/user-attachments/assets/c8af7a16-6b23-497d-b912-10588355c869" />

Decrease the spacing between Hero section and pagination bar
<img width="1905" height="541" alt="image" src="https://github.com/user-attachments/assets/58d38659-b1e1-4087-b407-9fe52714b530" />

Fixated the position of pagination bar for mobile layout
<img width="298" height="476" alt="image" src="https://github.com/user-attachments/assets/6031cb29-63e5-4adb-8587-d5e721fb348b" />

Buzzpy commented:
That's perfect. It's ready to be merged! Thanks for the contribution, @KC900201.
